File: count_lines.pl
use strict;
use warnings;
use List::Util qw(sum);

BEGIN {
    package Record;
    use Moo;
    has name => (is => 'rw');
    has count => ( is => 'rw');

    $INC{"Record.pm"};
}

use Record;

sub create_records {
    my $count = scalar(grep {index(lc substr($_, 1, 5), 'bc') >= 0} @_);
    # return {
    #     "name" => $_[0],
    #     "count" => $count
    # };
    return new Record->new(
        name => $_[0],
        count => $count
    );
}

sub main {
    my @records;
    while(my $line = <STDIN>) {
       push @records, create_records(split(/\t/, $line));
    }
    print sum(map {$_->{count}} @records) . "\n";
}
main();

File: count_lines.py
import sys

class Record(object):
    __slots__ = ["name", "count"]
    def __init__(self, name, count):
        self.name = name
        self.count = count

def create_record(vals):
    count = len([val for val in vals if "bc" in val[1:4].lower()])
    record = Record(vals[0], count)
    # record = {"name": vals[0], "count": count}
    return record

def main():
    records = []
    for line in sys.stdin:
        records.append(create_record(line.split('\t')))

    print(sum([r.count for r in records]))

if __name__ == '__main__':
    main()